xapian-core/common/serialise-double.cc

```diff
@@ -38,175 +40,153 @@
 
 using namespace std;
 
-// The serialisation we use for doubles is inspired by a comp.lang.c post
-// by Jens Moeller:
-//
-// https://groups.google.com/group/comp.lang.c/browse_thread/thread/6558d4653f6dea8b/75a529ec03148c98
-//
-// The clever part is that the mantissa is encoded as a base-256 number which
-// means there's no rounding error provided both ends have FLT_RADIX as some
-// power of two.
-//
-// FLT_RADIX == 2 seems to be ubiquitous on modern UNIX platforms, while
-// some older platforms used FLT_RADIX == 16 (IBM machines for example).
-// FLT_RADIX == 10 seems to be very rare (the only instance Google finds
-// is for a cross-compiler to some TI calculators).
-
-#if FLT_RADIX == 2
-# define MAX_MANTISSA_BYTES ((DBL_MANT_DIG + 7 + 7) / 8)
-# define MAX_EXP ((DBL_MAX_EXP + 1) / 8)
-# define MAX_MANTISSA (1 << (DBL_MAX_EXP & 7))
-#elif FLT_RADIX == 16
-# define MAX_MANTISSA_BYTES ((DBL_MANT_DIG + 1 + 1) / 2)
-# define MAX_EXP ((DBL_MAX_EXP + 1) / 2)
-# define MAX_MANTISSA (1 << ((DBL_MAX_EXP & 1) * 4))
-#else
-# error FLT_RADIX is a value not currently handled (not 2 or 16)
-// # define MAX_MANTISSA_BYTES (sizeof(double) + 1)
-#endif
+#ifdef FOLLOWS_IEEE
 
-static int base256ify_double(double &v) {
-    int exp;
-    v = frexp(v, &exp);
-    // v is now in the range [0.5, 1.0)
-    --exp;
-#if FLT_RADIX == 2
-    v = scalbn(v, (exp & 7) + 1);
-#else
-    v = ldexp(v, (exp & 7) + 1);
-#endif
-    // v is now in the range [1.0, 256.0)
-    exp >>= 3;
-    return exp;
+string serialise_double(double v)
+{
+# ifdef WORDS_BIGENDIAN
+    uint64_t temp;
+    static_assert(sizeof(temp) == sizeof(v));
```

Comment:
Omitting the second argument to `static_assert()` requires C++17, and currently we target C++11.  So can you add a description of what's being asserted here?